src/sfError.ts

```diff
@@ -59,13 +72,17 @@ export class SfError<T = unknown> extends NamedError {
    */
   public constructor(
     message: string,
-    name?: string,
+    name = 'SfError',
     actions?: string[],
     exitCodeOrCause?: number | Error,
-    cause?: Error
+    cause?: unknown
   ) {
-    cause = exitCodeOrCause instanceof Error ? exitCodeOrCause : cause;
-    super(name ?? 'SfError', message || name, cause);
+    if (typeof cause !== 'undefined' && !(cause instanceof Error)) {
+      throw new TypeError(`The cause, if provided, must be an instance of Error. Received: ${typeof cause}`);
+    }
```

Comment:
related to the WI about correctly bubbling up the erro from iso-git > STL > SDR:

STL throws an `SfError` with data in it, SDR will wrap it and rethrow (new SfError...) but you have to remember to call `error.setData(originalError)` or the data is lost. 
Could we check here in the constructor if `cause instanceof SfError` and save possible `cause.data` in the new error?

Reply:
a "parent" error could call `setData` or mutating the data prop and losing the `cause` data.

but, it's a good idea.

Reply:
but maybe this is the least bad option. 
```
 this.data = resolvedCause.data as T;
```